f8a_report/graph_report_generator.py

```diff
@@ -0,0 +1,221 @@
+                "actual_latest_version": latest
+            }
+            report_result[eco + ":" + pkg] = tmp
+
+    return report_result
+
+
+def execute_gremlin_dsl(payload, url=GREMLIN_SERVER_URL_REST):
+    """Execute the gremlin query and return the response."""
+    try:
+        response = get_session_retry().post(url, json=payload)
+        if response.status_code == 200:
+            return response.json()
+        else:
+            _logger.error(
+                "HTTP error {code}. Error retrieving data from {url}.".format(
+                    code=response.status_code, url=url))
+            return None
+
+    except Exception:
+        _logger.error(traceback.format_exc())
+        return None
+
+
+def get_session_retry(retries=3, backoff_factor=0.2, status_forcelist=(404, 500, 502, 504),
+                      session=None):
+    """Set HTTP Adapter with retries to session."""
+    session = session or requests.Session()
+    retry = Retry(total=retries, read=retries, connect=retries,
+                  backoff_factor=backoff_factor, status_forcelist=status_forcelist)
+    adapter = HTTPAdapter(max_retries=retry)
+    session.mount('http://', adapter)
+    return session
+
+
+def get_response_data(json_response, data_default):
+    """Retrieve data from the JSON response.
+
+    Data default parameters takes what should data to be returned.
+    """
+    return json_response.get("result", {}).get("data", data_default)
+
+
+def batch_query_executor(query_string, args):
+    """Execute the gremlin query in batches of 50."""
+    query = "epv=[];"
+    tmp_query = ""
+    counter = 0
+    result_data = []
+    for arg in args:
+        if len(arg) == 2:
+            tmp_query = query_string.format(arg0=arg['0'], arg1=arg['1'])
+            counter += 1
+        elif len(arg) == 3:
+            tmp_query = query_string.format(arg0=arg['0'], arg1=arg['1'], arg2=arg['2'])
+            counter += 1
+        if counter == 1:
+            query = "epv=[];"
+        query += tmp_query
+
+        if counter >= GREMLIN_QUERY_SIZE:
+            counter = 0
+            payload = {'gremlin': query}
+            gremlin_response = execute_gremlin_dsl(payload)
+            if gremlin_response is not None:
+                result_data += get_response_data(gremlin_response, [{0: 0}])
+            else:
+                _logger.error("Error while trying to fetch data from graph. "
+                              "Expected response, got None...Query->", query)
+
+    if counter < GREMLIN_QUERY_SIZE:
+        payload = {'gremlin': query}
+        gremlin_response = execute_gremlin_dsl(payload)
+        if gremlin_response is not None:
+            result_data += get_response_data(gremlin_response, [{0: 0}])
+        else:
+            _logger.error("Error while trying to fetch data from graph. "
+                          "Expected response, got None...Query->", query)
+
+    return result_data
+
+
+"""
```

Comment:
Please remove this commented out block. This need not go to the master branch.

Reply:
removed